ygen/enumgen_test.go

```diff
@@ -907,6 +911,341 @@ func TestFindEnumSet(t *testing.T) {
 				},
 			},
 		},
+	}, {
+		name: "two enums with deduplication disabled, where duplication of enums is only happening for uncompressed due to compressed context being the same (i.e. config/state)",
+		in: map[string]*yang.Entry{
+			"/container/config/enumeration-leaf": {
+				Name: "enumeration-leaf",
+				Type: &yang.YangType{
+					Name: "enumeration",
+					Enum: &yang.EnumType{},
+				},
+				Node: &yang.Enum{
+					Name: "enumeration-leaf",
+					Parent: &yang.Grouping{
+						Name: "foo",
+						Parent: &yang.Module{
+							Name: "base-module2",
+						},
+					},
+				},
+				Parent: &yang.Entry{
+					Name: "config",
+					Parent: &yang.Entry{
+						Name:   "container",
+						Parent: &yang.Entry{Name: "base-module2"},
+					},
+				},
+			},
+			"/container/state/enumeration-leaf": {
+				Name: "enumeration-leaf",
+				Type: &yang.YangType{
+					Name: "enumeration",
+					Enum: &yang.EnumType{},
+				},
+				Node: &yang.Enum{
+					Name: "enumeration-leaf",
+					Parent: &yang.Grouping{
+						Name: "foo",
+						Parent: &yang.Module{
+							Name: "base-module2",
+						},
+					},
+				},
+				Parent: &yang.Entry{
+					Name: "state",
+					Parent: &yang.Entry{
+						Name: "container",
+						Parent: &yang.Entry{
+							Name: "base-module2",
+						},
+					},
+				},
+			},
+		},
+		inSkipEnumDeduplication: true,
+		wantCompressed: map[string]*yangEnum{
+			"BaseModule2_Container_EnumerationLeaf": {
+				name: "BaseModule2_Container_EnumerationLeaf",
+				entry: &yang.Entry{
+					Name: "enumeration-leaf",
+					Type: &yang.YangType{
+						Enum: &yang.EnumType{},
+					},
+				},
+			},
+		},
+		wantUncompressed: map[string]*yangEnum{
+			"BaseModule2_Container_State_EnumerationLeaf": {
+				name: "BaseModule2_Container_State_EnumerationLeaf",
+				entry: &yang.Entry{
+					Name: "enumeration-leaf",
+					Type: &yang.YangType{
+						Enum: &yang.EnumType{},
+					},
+				},
+			},
+			"BaseModule2_Container_Config_EnumerationLeaf": {
+				name: "BaseModule2_Container_Config_EnumerationLeaf",
+				entry: &yang.Entry{
+					Name: "enumeration-leaf",
+					Type: &yang.YangType{
+						Enum: &yang.EnumType{},
+					},
+				},
+			},
+		},
+	}, {
+		name: "two enums with deduplication disabled, and where duplication occurs for both compressed and decompressed",
```

Comment:
(comment) This is an interesting corner-case, since in a compressed schema, this enum type would never actually be referenced. For example:

```yang
grouping foo-config {
   leaf a { type enumeration { enum ONE; } }
}

container foo {
   container config { uses foo-config; }
   contianer state { uses foo-state; }
}
```

In this case, compression will remove `/foo/state/a` but `/foo/state/a` will still create `Foo_State_A` in the output code.

I don't think we need to worry about this - because we're explicitly doing what the user asked, it just doesn't really make a huge amount of sense :-)